### ISSUE TITLE
Linting is failing after upgrade

### ISSUE THREAD

**Issue by kaiyoma** [SublimeLinter/SublimeLinter#1103]
SublimeLinter upgraded automatically for me recently and now I'm seeing these stack traces over and over in the console:

```
Traceback (most recent call last):
  File "C:\Program Files\Sublime Text\sublime_plugin.py", line 496, in on_modified_async
    callback.on_modified_async(v)
  File "sublime_linter in ...\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 104, in on_modified_async
  File "lint.settings in ...\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 33, in get
  File "lint.settings in ...\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 25, in settings
AttributeError: 'NoneType' object has no attribute 'load_settings'
Traceback (most recent call last):
  File "C:\Program Files\Sublime Text\sublime_plugin.py", line 530, in on_activated_async
    callback.on_activated_async(v)
  File "sublime_linter in ...\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 118, in on_activated_async
  File "lint.settings in ...\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 33, in get
  File "lint.settings in ...\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 25, in settings
AttributeError: 'NoneType' object has no attribute 'load_settings'
```

Linting just stopped working today. Any idea what's going on?

**Comment by kaste** [comment]
Please restart Sublime.

**Comment by Djspaceg** [comment]
I'm actually seeing `"NoneType"` error logs in my Sublime Text 3 console too. It's not exactly the same, because I'm on mac, but maybe my stacktrace will be more helpful to find the problem?
```
Traceback (most recent call last):
  File "/Applications/Sublime Text.app/Contents/MacOS/sublime_plugin.py", line 389, in run_callback
    expr()
  File "/Applications/Sublime Text.app/Contents/MacOS/sublime_plugin.py", line 473, in <lambda>
    run_callback('on_post_save', callback, lambda: callback.on_post_save(v))
  File "/Users/blake/Library/Application Support/Sublime Text 3/Packages/Gulp/event_task.py", line 16, in on_post_save
    settings = Settings()
  File "/Users/blake/Library/Application Support/Sublime Text 3/Packages/Gulp/settings.py", line 33, in __init__
    self.sources = [ProjectData(), self.user_settings]
  File "/Users/blake/Library/Application Support/Sublime Text 3/Packages/Gulp/settings.py", line 56, in __init__
    self._project_data = active_window.project_data().get(Settings.PACKAGE_NAME, {})
AttributeError: 'NoneType' object has no attribute 'get'
reloading settings Packages/User/SublimeLinter.sublime-settings
```

What's interesting to me about this is that this error occurs *only* after I generate a `SublimeLinter.sublime-settings` file. If there's no `SublimeLinter.sublime-settings` file, no error. The contents of the file seem to also be irrelevant, as even the default-generated file throws this error.
```
// SublimeLinter Settings - User
{

}
```
Removing the comment, removing the trailing newline, adding properties to the object doesn't resolve the issue.

I've restarted Sublime Text multiple times. I've removed old linters and am now only have `SublimeLinter-eslint` enabled, which I've removed and reinstalled. My paths appear to be correct, I've updated everything I can think of and I do actually get basic linting output in the editor, and even the tooltip on error-hover.

It's just something about reading/parsing/loading the user settings file, whether it's self/auto-generated or manually created.

**Comment by kaiyoma** [comment]
@kaste Fixed my issue. Thanks a ton for your quick response! Feel free to close this or maybe keep it open for @Djspaceg's issue.

**Comment by Djspaceg** [comment]
Nah, it looks like that specific error I was getting is related to `Gulp`, and not directly with `SublimeLinter`. I'm still having a ton of other issues trying to get the gutter theme back, but this error specifically looks unrelated.

Safe to close, unless you want to use it to further investigate the Gulp interaction issue.

**Comment by kaste** [comment]
Gulp is unrelated. @kaiyoma I'm interested if you get this after *every* update we ship in the future.

**Comment by kaiyoma** [comment]
@kaste If I see any future issues, I'll definitely let you guys know.

**Comment by raniesantos** [comment]
![screenshot from 2018-03-06 14-54-07](https://user-images.githubusercontent.com/8528269/37018166-51f45b8c-214e-11e8-8475-6c67b57a41d2.png)

**Comment by kaiyoma** [comment]
@kaste I saw issues again today after opening Sublime and receiving the 4.0.3 update. I had to shut down and restart three times before linting worked correctly with no errors in the console.

**Comment by kaste** [comment]
Three restarts? Maybe you had other updates as well.

Can you post the detailed logs (obviously the next time) so we can actually look at the load/reload order?

**Comment by kaiyoma** [comment]
No, there weren't any other updates. Actually, I just ran into more problems today, even though there weren't any updates. After I had been working for a little bit, I noticed that linting wasn't working again. When I checked the console, I saw the same Python errors I posted above. Closing and restarting fixes the issue, but I'm finding I have to do this at least once a day. Something about the new version isn't working quite right.
